birdhouse/deployment/deploy-data-specific-image

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+# Deploy data from git repo(s) to local folder(s).
+# This script is run directly on a specific image (such as pavics/crim-jupyter-eo or pavics/crim-jupyter-nlp).
+# It will be used to download and update the different tutorial notebooks associated with a specific image.
+#
+# This is meant to be run on the same host running PAVICS.
+#
+# The data details is specified in a yaml config file (TEMPLATE_CONFIG_YML), using this format :
+# - repo_url: path to the github repo (ex.: https://github.com/crim-ca/pavics-jupyter-images)
+#   branch: name of the branch containing the required version of the data
+#   source_path: path of the desired file or folder in the source repo
+#   dest_dir: directory where the data will be copied to in the specific image where the script is run
+#
+# One or more entries respecting this format can be added to the yaml file, in order to download multiple file/folders.
+#
+# An installation of jq and yq is preferred prior to executing this script.
+#
+# Setting environment variable DEPLOY_DATA_LOGFILE='/path/to/logfile.log'
+# will redirect all STDOUT and STDERR to that logfile so this script will be
+# completely silent.
+
+if [ ! -z "$DEPLOY_DATA_LOGFILE" ]; then
+    exec >>$DEPLOY_DATA_LOGFILE 2>&1
+fi
+
+START_TIME="`date -Isecond`"
+echo "==========
+deploy-data-specific-image START_TIME=$START_TIME"
+
+TEMPLATE_CONFIG_YML="$1"
+if [ -z "$TEMPLATE_CONFIG_YML" ]; then
+    echo "ERROR: missing config.yml file" 1>&2
+    exit 2
+fi
+
+# Empty value could mean typo in the keys in the config file.
+ensure_not_null() {
+    if [ "$*" = null ]; then
+        echo "ERROR: value empty" 1>&2
+        exit 1
+    fi
+}
+
+# Check installation of required package to read yaml files
+if ! command -v jq &> /dev/null; then
+  apt-get -y install jq
+fi
+if ! command -v yq &> /dev/null; then
+  pip install yq
+fi
```

Comment:
I would suggest not to install packages within the actual deploy script.
Instead, printing an issue and prematurely exiting the script would make sense. 
A clear separation between the requirements and actual logic is preferred.